sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -915,6 +915,52 @@ __ESIMD_API void lsc_block_store(T *p, __ESIMD_NS::simd<T, NElts> vals) {
                                               vals.data());
 }
 
+namespace detail {
+// Compile-time checks for lsc_load2d/store2d restrictions.
+template <typename T, int BlockWidth, int BlockHeight, int NBlocks,
+          bool Transposed, bool Transformed, bool IsStore = false>
+constexpr void check_lsc_block_2d_restrictions() {
+  constexpr int GRFByteSize = BlockWidth * BlockHeight * NBlocks * sizeof(T);
+  static_assert(!IsStore || GRFByteSize <= 512,
+                "2D store supports 512 bytes max");
+  static_assert(IsStore || GRFByteSize <= 2048,
+                "2D load supports 2048 bytes max");
+  static_assert(!Transposed || !Transformed,
+                "Transposed and transformed is not supported");
+  static_assert(!Transposed || (Transposed && NBlocks == 1),
+                "Transposed expected to be 1 block only");
+  static_assert(!Transformed || (sizeof(T) == 1 || sizeof(T) == 2),
+                "VNNI transform is supported only for data size u8 or u16");
+  static_assert(!Transposed || (sizeof(T) == 4 || sizeof(T) == 8),
+                "Transposed load is supported only for data size u32 or u64");
```

Comment:
Just wanted to check. Transposed load handles 4 and 8 byte elements. Is transposed _store_ works for all data sizes?